### PR TITLE
Doctor: add check for .git folder

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

Just a simple check for `.git`. Emits a warning if absent, but maybe it should be an error, since we need the submodules to build?

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
`C901 'doctor' is too complex (17)`

😢

File: lib/python/qmk/cli/doctor.py [hunk 1]
```diff
@@ -154,6 +154,17 @@ def check_submodules():
...
 
+def check_git_repo():
+    """Checks that the .git directory exists inside QMK_HOME.
+
+    This is a decent enough indicator that the qmk_firmware directory is a
+    proper Git repository, rather than a .zip download from GitHub.
+    """
+    dot_git_dir = QMK_FIRMWARE / '.git'
+
+    return CheckStatus.OK if dot_git_dir.exists() else CheckStatus.WARNING
```
Comment (Erovia) [inline lib/python/qmk/cli/doctor.py:165]:
```suggestion
    return CheckStatus.OK if dot_git_dir.is_dir() else CheckStatus.WARNING
```
If we want to be more pedantic.
This also checks existence.

**Review by Erovia — APPROVED** [review]

**Comment by Erovia** [comment]
> `C901 'doctor' is too complex (17)`
> 
> cry

I think we could move the checks to OS-specific modules.
But that may be better in a dedicated PR.

**Comment by skullydazed** [comment]
> C901 'doctor' is too complex (17)

I think #11242 should fix you up.

**Review by Erovia — APPROVED** [review]

**Pull request merged**